lib/stripe_tester.rb

```diff
@@ -125,6 +125,18 @@ def self.remove_url
     @url = nil
   end
 
+  def self.webhook_password=(webhook_password)
+    @webhook_password =  webhook_password
+  end
+
+  def self.webhook_password
+    @webhook_password
+  end
+
+  def self.basic_authentication_password
+    @webhook_password || (self.webhook_url ? self.webhook_url.password : nil)
```

Comment:
There is no need to check if the `webhook_url` exists, since it raises an error when it's not set.

Test: https://github.com/amratab/stripe_tester/blob/authentication_support/spec/stripe_tester_spec.rb#L164

Suggested refactor:

```ruby
@webhook_password || self.webhook_url.password
```